src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -35,95 +33,1003 @@ const ActorInit En_Sw_InitVars = {
+        pos.z = this->actor.world.pos.z + (Math_CosS(angle) * 2.0f);
+        func_8002836C(globalCtx, &pos, &velocity, &accel, &primColor, &envColor, 20, 30, 12);
+    }
+}
+
+void func_80B0D14C(EnSw* this, GlobalContext* globalCtx, s32 cnt) {
+    Color_RGBA8 primColor = { 80, 80, 50, 255 };
+    Color_RGBA8 envColor = { 100, 100, 80, 0 };
+    Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f accel = { 0.0f, 0.300000011921f, 0.0f };
+    Vec3f pos;
+    s16 angle;
+    s32 i;
+
+    angle = (Rand_ZeroOne() - 0.5f) * 65536.0f;
+    for (i = cnt; i >= 0; i--, angle += (s16)(0x10000 / cnt)) {
+        accel.x = (Rand_ZeroOne() - 0.5f) * 2.0f;
+        accel.z = (Rand_ZeroOne() - 0.5f) * 2.0f;
+        pos.x = this->actor.world.pos.x + (Math_SinS(angle) * 14.0f);
+        pos.y = this->actor.world.pos.y;
+        pos.z = this->actor.world.pos.z + (Math_CosS(angle) * 14.0f);
+        func_8002836C(globalCtx, &pos, &velocity, &accel, &primColor, &envColor, 20, 40, 10);
+    }
+}
+
+void func_80B0D364(EnSw* this, GlobalContext* globalCtx) {
+    if (((this->actor.params & 0xE000) >> 0xD) == 4) {
+        this->unk_38C = 0;
+        this->actionFunc = func_80B0D3AC;
+    } else {
+        this->unk_38C = 10;
+        this->actionFunc = func_80B0D3AC;
+    }
+}
+
+void func_80B0D3AC(EnSw* this, GlobalContext* globalCtx) {
+    if (this->unk_38C != 0) {
+        if ((this->unk_38C & 4) != 0) {
+            func_80B0CF44(this, globalCtx, 5);
+        }
+        this->unk_38C--;
+        if (this->unk_38C == 0) {
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 0x28, NA_SE_EN_STALGOLD_UP_CRY);
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 0x28, NA_SE_EN_DODO_M_UP);
+        } else {
+            return;
+        }
+    }
+    Math_ApproachF(&this->actor.scale.x, 0.02f, 0.2f, 0.01f);
+    Actor_SetScale(&this->actor, this->actor.scale.x);
+    this->actor.world.pos.x += this->unk_364.x * this->actor.velocity.y;
+    this->actor.world.pos.y += this->unk_364.y * this->actor.velocity.y;
+    this->actor.world.pos.z += this->unk_364.z * this->actor.velocity.y;
+    this->actor.world.pos.x += this->unk_37C.x * this->actor.speedXZ;
+    this->actor.world.pos.y += this->unk_37C.y * this->actor.speedXZ;
+    this->actor.world.pos.z += this->unk_37C.z * this->actor.speedXZ;
+    this->actor.velocity.y += this->actor.gravity;
+    this->actor.velocity.y = CLAMP_MIN(this->actor.velocity.y, this->actor.minVelocityY);
+    if (this->actor.velocity.y < 0.0f) {
+        this->unk_360 = 0;
+    }
+
+    if (func_80B0C0CC(this, globalCtx, 1) == 1) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+        func_80B0D14C(this, globalCtx, 8);
+        this->actor.scale.x = 0.02f;
+        Actor_SetScale(&this->actor, 0.02f);
+        this->actionFunc = func_80B0D590;
+        this->actor.velocity.y = 0.0f;
+        this->actor.speedXZ = 0.0f;
+        this->actor.gravity = 0.0f;
```

Comment:
chained assign?

Reply:
doesnt improve readability in this case imo